pandas/io/tests/test_pytables.py

```diff
@@ -4687,6 +4687,14 @@ def test_duplicate_column_name(self):
             other = read_hdf(path, 'df')
             tm.assert_frame_equal(df, other)
 
+    def test_preserve_timedeltaindex_type(self):
+        df = DataFrame(np.random.normal(size=(10,5)))
+        df.index = timedelta_range(start='0s',periods=10,freq='1s')
+
+        with ensure_clean_store(self.path) as store:
+            
+            store['df'] = df
+            self.assertEqual(type(store['df'].index), TimedeltaIndex)
 
 def _test_sort(obj):
```

Comment:
I don't recall exactly if the fixed stores preserve some meta data, e.g. ``name/freq``. So pls add the attributes (e.g. you can add them with ``timedelta_range(......, name='foo')`` and see if they are preserved (which the above testing method will catch).